src/Build/BuildCheck/Infrastructure/CheckWrapper.cs

```diff
@@ -52,10 +75,37 @@ internal void StartNewProject(
         }
     }
 
+    internal void ReportResult(BuildCheckResult result, ICheckContext checkContext, CheckConfigurationEffective config)
+    {
+        if (_reportsCountPerRule is not null)
+        {
+            if (!_reportsCountPerRule.ContainsKey(result.CheckRule.Id))
+            {
+                _reportsCountPerRule[result.CheckRule.Id] = 0;
+            }
+            _reportsCountPerRule[result.CheckRule.Id]++;
+
+            if (_reportsCountPerRule[result.CheckRule.Id] == MaxReportsNumberPerRule + 1)
+            {
+                checkContext.DispatchAsCommentFromText(MessageImportance.Normal, $"The check '{Check.FriendlyName}' has exceeded the maximum number of results allowed for the rule '{result.CheckRule.Id}'. Any additional results will not be displayed.");
```

Comment:
Let's have the message localizable and let's append it to the result.

Otherwise it can get lost (e.g. when users have terminal logger - which is default)

Reply:
Plus let's dismount the check - so that it doesn't receive and operate on other data (it's fine to dismount even if the check has multiple rules)

Reply:
You mean to append the message to the latest result that has been shown? I can do that, albeit it would not look very nice in code I am afraid. 

As for localization I suggest given that this will be appended to the latest build check result and they are not localized, not to localize it as well. Otherwise, it would look strange: part of the message is localized, part is not.